src/qibo/_openqasm.py

```diff
@@ -0,0 +1,190 @@
+import numpy as np
+import openqasm3
+
+import qibo
+from qibo.config import raise_error
+
+
+class DefinedGate:
```

Comment:
This is the class that we discussed it could be replaced by `FusedGate`, right?

Reply:
I am not sure this can be straight out replaced by `FusedGate`, but it could in principle construct a unique fused gate object rather than a list of simple gates.